lib/resourcemerge/apps.go

```diff
@@ -10,6 +10,11 @@ import (
 func EnsureDeployment(modified *bool, existing *appsv1.Deployment, required appsv1.Deployment) {
 	EnsureObjectMeta(modified, &existing.ObjectMeta, required.ObjectMeta)
 
+	if required.Spec.Replicas != nil && required.Spec.Replicas != existing.Spec.Replicas {
```

Comment:
Why do you think this is important?

Reply:
We are using this lib in machine-api-operator. When MAO reconciles its owned deployments, it is desired that replica count should also be reconciled. https://jira.coreos.com/browse/CLOUD-484